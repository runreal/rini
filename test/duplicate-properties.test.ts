import { assertSnapshot } from 'std/testing/snapshot.ts'
import * as path from 'std/path/mod.ts'
import * as ini from '../mod.ts'

const fixture = path.resolve('./test/fixtures/duplicate.ini')
const data = Deno.readTextFileSync(fixture)

function serializer(actual: string): string {
	return actual.replace(/\r\n/g, '\n').trim()
}

Deno.test('decode with duplicate properties', async function (t): Promise<void> {
	const d = ini.decode(data)
	await assertSnapshot(t, d)
})

Deno.test('encode with duplicate properties', async function (t): Promise<void> {
	const e = ini.encode({
		ar: ['1', '2', '3'],
		br: ['1', '2'],
	})
	await assertSnapshot(t, e, { serializer })
})

Deno.test('decode duplicate properties with bracketedArray=false', async function (t): Promise<void> {
	const d = ini.decode(data, { bracketedArray: false })
	await assertSnapshot(t, d)
})

Deno.test('encode duplicate properties with bracketedArray=false', async function (t): Promise<void> {
	const e = ini.encode({
		ar: ['1', '2', '3'],
		br: ['1', '2'],
	}, { bracketedArray: false })
	await assertSnapshot(t, e, { serializer })
})
