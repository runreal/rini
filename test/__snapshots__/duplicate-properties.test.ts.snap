export const snapshot = {}

snapshot[`decode with duplicate properties 1`] = `
[Object: null prototype] {
  ar: [
    "one",
    "three",
  ],
  brr: "3",
  str: "3",
  zr: [
    "deedee",
    "123",
  ],
}
`

snapshot[`encode with duplicate properties 1`] = `
ar[]=1
ar[]=2
ar[]=3
br[]=1
br[]=2
`

snapshot[`decode duplicate properties with bracketedArray=false 1`] = `
[Object: null prototype] {
  "ar[]": [
    "one",
    "three",
  ],
  "zr[]": "deedee",
  brr: [
    "1",
    "2",
    "3",
    "3",
  ],
  str: "3",
  zr: "123",
}
`

snapshot[`encode duplicate properties with bracketedArray=false 1`] = `
ar=1
ar=2
ar=3
br=1
br=2
`
